integration/smartbft/smartbft_block_deliverer_test.go

```diff
@@ -0,0 +1,205 @@
+)
+
+func smartBftBlockDelivererTest(
+	testDir string,
+	client *docker.Client,
+	network *nwo.Network,
+	networkProcess ifrit.Process,
+	ordererProcesses []ifrit.Process,
+	peerProcesses ifrit.Process) {
+
+	networkProcess = nil
+	ordererProcesses = nil
+	peerProcesses = nil
+	var err error
+
+	/* Create a client*/
+	client, err = docker.NewClientFromEnv()
+
+	Expect(err).NotTo(HaveOccurred())
+
+	/* Create networkConfig for multiple orderers */
+	networkConfig := nwo.MultiNodeSmartBFT()
+	networkConfig.Channels = nil
+
+	channel := "testchannel1"
+
+	fmt.Println(client.Endpoint())
+
+	/* Create a network from the networkConfig */
+	startPort := StartPort()
+	print(startPort)
+	network = nwo.New(networkConfig, testDir, client, startPort, components)
+	network.GenerateConfigTree()
+	network.Bootstrap()
+
+	var ordererRunners []*ginkgomon.Runner
+	for _, orderer := range network.Orderers {
+		runner := network.OrdererRunner(orderer)
+		runner.Command.Env = append(runner.Command.Env, "FABRIC_LOGGING_SPEC=orderer.consensus.smartbft=debug:grpc=debug")
+		ordererRunners = append(ordererRunners, runner)
+		proc := ifrit.Invoke(runner)
+		ordererProcesses = append(ordererProcesses, proc)
+		Eventually(proc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+	}
+
+	joinChannel(network, channel)
+
+	By("Waiting for followers to see the leader")
+	Eventually(ordererRunners[1].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+	Eventually(ordererRunners[2].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+	Eventually(ordererRunners[3].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+
+	/* Create a stream with client for each orderer*/
+	var allStreams []ordererProtos.AtomicBroadcast_BroadcastClient
+	for _, o := range network.Orderers {
+		conn := network.OrdererClientConn(o)
+		broadcaster, err := ordererProtos.NewAtomicBroadcastClient(conn).Broadcast(context.Background())
+		Expect(err).ToNot(HaveOccurred())
+		allStreams = append(allStreams, broadcaster)
+	}
+
+	/* Fill the ledger with blocks */
+	By("Filling ledger with blocks")
+	closeTxProducer := make(chan bool)
+	go func() {
+		defer GinkgoRecover()
+		var i uint64
+		for {
+			buff := make([]byte, 8)
+			binary.BigEndian.PutUint64(buff, i)
+
+			/* Create transaction envelope which will be broadcast to all orderers */
+			env := ordererclient.CreateBroadcastEndorserTransactionEnvelope(network, network.Orderers[1], channel, buff)
+
+			for _, stream := range allStreams {
+				err := stream.Send(env)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			i++
+
+			finishSending := false
+			select {
+			case <-closeTxProducer:
+				finishSending = true
+			case <-time.After(time.Second):
+			}
+			if finishSending {
+				return
+			}
+		}
+	}()
+	Eventually(ordererRunners[1].Err(), 10*time.Second).Should(gbytes.Say("Delivering proposal, writing block 10"))
+	closeTxProducer <- true
```

Comment:
This may fill the orderer with way too many blocks. why not run this precisely the number of desired times "K"?
Moreover, since each orderer is committing asynch, you have to check that every orderer had arrive to K, or at least, check the correct runner (you check 1 but then grab the ledger from 0, below). I think it will be sfare to check all orderers had reached K.

Reply:
Done